app/models/request.rb

```diff
@@ -1,4 +1,20 @@
 class Request < ApplicationRecord
-  belongs_to :sender, class_name: 'User'
+  # callbacks
+  
+
+  # Validations
+  validates :sender_id, uniqueness: { scope: :receiver_id }
+  validate :sender_and_receiver_are_not_friends
+
+  # Associations
   belongs_to :receiver, class_name: 'User'
+  belongs_to :sender, class_name: 'User'
+
+  # Instance Methods
+  def sender_and_receiver_are_not_friends
```

Comment:
This `sender_and_receiver_are_not_friends' method isn't used anywhere else but in the validation above. So i recommend just making this a `private` method.

Further, please see my comment on `friendship.rb` for a suggested way to enforce uniqueness of record. We can apply the exact same principle here for `Request`.

Reply:
Hello @rvvergara
We have enforced uniqueness for both friendships and requests, but `sender_and_receiver_are_not_friends` ensures that any user can't request a friendship with any other user he is already friends with.